src/select.coffee

```diff
@@ -349,11 +368,19 @@ Ember.AddeparMixins.ResizeHandlerMixin, Ember.Widgets.KeyboardHelper,
     @set 'selection', content.findProperty(defaultPath)
   , 'content.[]'
 
-  selectableOptionsDidChange: Ember.observer ->
+  ###*
+  * Update highlighted item to the first option when the current highlighted
+  * item does not exist.
+  * @function
+  ###
+  updateHighlightedItem: ->
     if @get('showDropdown')
       highlighted = @get('highlighted')
       if not @get('selectableOptions').contains(highlighted)
         @set 'highlighted', @get('selectableOptions.firstObject')
+
+  selectableOptionsDidChange: Ember.observer ->
```

Comment:
1. I didn't put an attempt to fix all of the observers in this component in this PR. I happened to find some of the observers are straightforward to be remove so I remove them. 
2. The reason of this observer (as part of the original solution) was to tackle the problem that the content of the selection is changed, for which the current highlighted item does not exist anymore. 
It is not straightforward to build computed properties because here we need the highlighted index bound to the UI, while we need the highlighted item for other operations. With only highlighted index, we cannot know if the item is already removed or not. Something like that. I can explain in person when we want to remove this observer.